carbonmark-api/src/utils/helpers/carbonProjects.utils.ts

```diff
@@ -0,0 +1,81 @@
+import { compact, merge } from "lodash";
+import { filter, pipe } from "lodash/fp";
+import {
+  GetAllProjectsQuery,
+  ProjectContent,
+} from "src/.generated/types/carbonProjects.types";
+import { SetRequired } from "../../../../lib/utils/typescript.utils";
+import { arrayToMap } from "../array.utils";
+import { extract, notNil, selector } from "../functional.utils";
+import { gqlSdk } from "../gqlSdk";
+
+type Args = {
+  registry: string; // e.g VCS
+  registryProjectId: string; //e.g 1121
+};
+
+/**
+ * Generates a unique key for a project using its registry and id.
+ * @param {Project} project - The project.
+ * @param {string} project.id - The id of the project.
+ * @param {string} project.registry - The registry of the project.
+ * @returns {string} The generated key.
+ */
+const projectKey = ({ registry, registryProjectId }: CarbonProject) =>
+  `${registry}-${registryProjectId}`;
+
+export type CarbonProject = GetAllProjectsQuery["allProject"][number] & {
+  content?: ProjectContent;
+};
+/**
+ * Fetches a carbon project based on the provided registry and id.
+ * @param {Object} args - The arguments.
+ * @param {string} args.registry - The registry of the project (e.g., VCS).
+ * @param {string} args.id - The id of the project (e.g., 1121).
+ * @returns {Promise<CarbonProject|null>} The fetched project or null if not found.
+ */
+export const fetchCarbonProject = async (args: Args) => {
+  const [{ allProject }, { allProjectContent }] = await Promise.all([
+    gqlSdk.carbon_projects.getProject(args),
+    gqlSdk.carbon_projects.getProjectContent(args),
+  ]);
+
+  const project = allProject.at(0);
+  const content = allProjectContent.at(0);
+  const key = projectKey(args);
+
+  return {
+    ...project,
+    content,
```

Comment:
this field is being passed to the final api response, not sure if that was intentional but I would revert as we should avoid changing the API responses in this pr.

Likely breaks the frontend as its expecting `project.images` not `project.content.images`